src/main/java/com/synopsys/integration/detect/lifecycle/run/step/ContainerScanStepRunner.java

```diff
@@ -4,25 +4,92 @@
 import com.synopsys.integration.detect.lifecycle.run.operation.OperationRunner;
-import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioUtil;
+import com.synopsys.integration.detect.util.bdio.protobuf.DetectProtobufBdioHeaderUtil;
 import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.rest.response.Response;
+import com.synopsys.integration.util.NameVersion;
+
+import net.minidev.json.JSONObject;
 
 public class ContainerScanStepRunner {
 
     private final OperationRunner operationRunner;
     private UUID scanId;
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+    private final NameVersion projectNameVersion;
+    private final BlackDuckRunData blackDuckRunData;
+    private final File binaryRunDirectory;
 
-    public ContainerScanStepRunner(OperationRunner operationRunner) throws DetectUserFriendlyException {
+    public ContainerScanStepRunner(OperationRunner operationRunner, NameVersion projectNameVersion, BlackDuckRunData blackDuckRunData) {
         this.operationRunner = operationRunner;
+        this.projectNameVersion = projectNameVersion;
+        this.blackDuckRunData = blackDuckRunData;
+        binaryRunDirectory = operationRunner.getDirectoryManager().getBinaryOutputDirectory();
     }
 
-    public UUID initiateScan(BlackDuckRunData blackDuckRunData) throws IOException, IntegrationException {
-        DetectProtobufBdioUtil detectProtobufBdioUtil = new DetectProtobufBdioUtil(UUID.randomUUID().toString(), "CONTAINER");
-        File bdioHeaderFile = detectProtobufBdioUtil.createProtobufBdioHeader();
-        scanId = operationRunner.uploadBdioHeaderToInitiateScan(blackDuckRunData, bdioHeaderFile);
+    public UUID invokeContainerScanningWorkflow() throws IntegrationException, IOException {
+        initiateScan();
+        uploadImageToStorageService();
+        uploadImageMetadataToStorageService();
         return scanId;
     }
+
+    public void initiateScan() throws IOException, IntegrationException {
+//        File bdioHeaderFile = new File(Application.class.getResource("/test-inputs/bdio-header.pb").getPath()); // temporary
+        DetectProtobufBdioHeaderUtil detectProtobufBdioHeaderUtil = new DetectProtobufBdioHeaderUtil(UUID.randomUUID().toString(), "CONTAINER", projectNameVersion);
+        File bdioHeaderFile = detectProtobufBdioHeaderUtil.createProtobufBdioHeader(binaryRunDirectory);
+        scanId = operationRunner.uploadBdioHeaderToInitiateScan(blackDuckRunData, bdioHeaderFile);
+        String scanIdString = scanId.toString();
+        logger.debug("Scan initiated with scan service. Scan ID received: {}", scanIdString);
+    }
+
+    public void uploadImageToStorageService() throws IntegrationException {
+        File containerImage = operationRunner.getContainerScanImage();
+        String storageServiceEndpoint = "/api/storage/containers/" + scanId;
```

Comment:
Use string .concat in all places.

Reply:
Used HttpUrl class's `appendRelativeUrl` method to join 2 paths as mentioned above